tests/db/compact_db_migrations.py

```diff
@@ -0,0 +1,99 @@
+"""
+    Copyright 2023 Inmanta
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Contact: code@inmanta.com
+
+
+    Tool to compact db migration scripts
+"""
+import asyncio
+import importlib
+import os
+import re
+
+import asyncpg
+import pytest
+
+UP_TO_VERSION = "202206290"  # Specify the version up to which migrations should be applied
+
+MIGRATIONS_DIR = "../../src/inmanta/db/versions"
+# Regular expression to match migration filenames
+MIGRATION_FILE_PATTERN = re.compile(r"^v(\d+)\.py$")
+
+
+# Extract version number from the filename
+def extract_version(filename):
+    match = MIGRATION_FILE_PATTERN.match(filename)
+    return int(match.group(1)) if match else None
+
+
+# Apply a migration script
+async def apply_migration(connection, migration_file_path):
+    spec = importlib.util.spec_from_file_location("migration_module", migration_file_path)
+    migration_module = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(migration_module)
+    if hasattr(migration_module, "update"):
+        await migration_module.update(connection)
+
+
+# Apply all migrations up to a specified version
+async def apply_migrations_up_to_version(connection, up_to_version):
+    migration_files = [f for f in os.listdir(MIGRATIONS_DIR) if MIGRATION_FILE_PATTERN.match(f)]
+    sorted_files = sorted(migration_files, key=extract_version)
+
+    for filename in sorted_files:
+        if extract_version(filename) < int(up_to_version):
+            print(f"Applying migration: {filename}")
+            migration_file_path = os.path.join(MIGRATIONS_DIR, filename)
+            await apply_migration(connection, migration_file_path)
```

Comment:
We should rely on our existing database migration tool `DBSchema` to apply the database migration scripts. Otherwise we risk that the existing database migration tool and this tool behave slightly differently, causing nasty bugs.